test/components/all_components_test.rb

```diff
@@ -0,0 +1,59 @@
+require "test_helper"
+
+class AllComponentsTest < ActionController::TestCase
+  Dir.glob("app/views/components/*.erb").each do |filename|
+    template = filename.split("/").last
+    component_name = template.sub("_", "").sub(".html", "").sub(".erb", "").gsub("-", "_")
+
+    context component_name do
+      yaml_file = "#{__dir__}/../../app/views/components/docs/#{component_name}.yml"
+
+      should "is documented" do
+        assert File.exist?(yaml_file)
+      end
+
+      should "have the correct documentation" do
+        # Psych 4's YAML.unsafe_load_file === Psych 3's YAML.load_file
+        # but until we drop support for Ruby 2.7 and Ruby 3.0, we need to support both major versions of Psych
```

Comment:
Since this app uses Ruby 3.2, is this relevant?

See commit: https://github.com/alphagov/frontend/commit/4ef8c1a62513e2888012243d366efdb23d99f38b

Reply:
Just giving it a try... was unsure since it said support for Ruby 2.7 `and` Ruby 3, but seems to pass locally with the simpler version.

Reply:
Aaand it's worked. Thanks for the prompt. I'll update the other PRs as well 👍

Reply:
Great, thanks @andysellick! 👍